### PR TITLE
Add allora_research_model_skills bundle

### PR DESCRIPTION
**Pull request by jmdkastro**
## Summary

Adds a new top-level folder `allora_research_model_skills/` containing three Claude Code skills (plus shared methodology docs) that guide builders through constructing financial prediction models from three different starting points:

- **hypothesis-driven** — deductive: start from a theory, build features to test it
- **signal-discovery** — inductive: start from data, discover what is predictable
- **robustness-first** — adversarial: start from validation gates, work backwards

`shared/` contains the common nine-principle methodology, feature engineering, loss design, and validation framework documentation that all three skills build on. The top-level `README.md` explains when to pick which skill.

## Why a separate top-level folder

The existing `skills/` folder hosts single-skill packages (`allora-data-exploration`, `allora-model-builder`, `allora-worker-manager`). This contribution is a curated bundle of three sibling methodology skills with their own README and shared docs, so it lives at the top level alongside `skills/` rather than inside it. Open to moving it under `skills/` if reviewers prefer.

## Excluded

The source folder includes a `test/` subdirectory with internal dry-run output and a test scenario; that's deliberately not copied — it's internal validation, not user-facing material.

## Test plan

- [x] Documentation-only addition; no runtime code changed
- [x] Confirm files render correctly on GitHub
- [x] Confirm CI is green (no path-restricted workflow should be triggered by docs-only changes)

---
## Summary by cubic
Adds the `allora_research_model_skills/` bundle with three methodology-first model-building skills and shared docs. Updates the root `README.md` with Model creation skills, a learning problem primer, TOC/diagram, and a small docs cleanup; docs-only, no runtime code.

- **New Features**
  - Bundle: three skills — `forge-hypothesis-driven`, `forge-signal-discovery`, `forge-robustness-first`; bundle `README.md` explains when to use each.
  - Shared docs: `methodology.md` (nine principles), `feature-engineering.md`, `loss-design.md`, `validation-framework.md`.
  - Root `README.md`: adds “Model creation skills” with a selection table and links; adds “The learning problem” primer; updates Contents TOC and adds an ASCII submission→evaluation→scores diagram; aligns Python prerequisite to 3.10+; removes a duplicate skills list in “The learning problem” and links to the new section instead.

<sup>Written for commit 438573618ec136c77a698a6ade2448b039954903. Summary will update on new commits. <a href="https://cubic.dev/pr/allora-network/allora-forge-builder-kit/pull/25?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: allora_research_model_skills/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+**For:** Builders who have been burned by overfitting, or who simply want the highest confidence that their model is real before deploying capital.
+
+**Approach:** Adversarial. You start by defining your validation framework and quality gates, then work backwards — designing features, architecture, and loss function specifically to survive the tests you already set up.
+
+**Best when:** You prioritize generalization over raw backtest performance. You are willing to build a simpler model if it means higher confidence that it works in production.
+
+## Prerequisites
+
+- **Python 3.9+**
```
[the remark this answers is not in the document]
Comment (jmdkastro) [reply allora_research_model_skills/README.md]: Addressed in c4c5e6c — bumped to 3.10+ to match the host repo's `pyproject.toml` (`requires-python = ">=3.10"`).

File: allora_research_model_skills/shared/loss-design.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+    d_true = z_true + s * delta
+    d_pred = z_pred + s_pred * delta
+
+    h1 = d2p1 * double_derivative(delta)
+    h3 = d2p1 * double_derivative(d_true)
+
+    # Region 1: opposite sign (u <= 0)
+    # Pseudo gradient for numerical stability:
+    G1 = h1 * z_pred - np.sign(z_true)
```
[the remark this answers is not in the document]
Comment (jmdkastro) [reply allora_research_model_skills/shared/loss-design.md:246]: Not acting on this. The mechanical observation is correct — `s` is regularized to 1 at `z_true == 0` while `np.sign(z_true)` returns 0 — but the practical impact is nil and changing it would silently alter the loss.

Region 1 is `u ≤ 0` where `u = s * z_pred`. At `z_true == 0`, `s = 1`, so we only enter Region 1 when `z_pred ≤ 0` — the corner of a piecewise loss. For continuous log-return targets this is a measure-zero event; on grid data it's an isolated sample at most. LightGBM sees a single off-by-1 in the gradient for that one sample and training behavior is unaffected.

The asymmetry between `s` (regularized for safe division/multiplication) and raw `np.sign` in the constant gradient term `G1 = h1 * z_pred - sign(z_true)` matches the original CZAR derivation. `s` exists to make `u = s * z_pred` and `d_true = z_true + s * delta` well-defined; the gradient's constant term is `-sign(z_true)` from the loss derivation itself. Substituting `s` would change the gradient at `z_true == 0` from 0 to ±1, which is not what the loss is supposed to do at that point.

File: allora_research_model_skills/robustness-first/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+## Phase 5: Execute Three-Stage Validation
+
+Run the pipeline through the three-stage methodology. This is where the builder's design is tested against reality.
+
+### Step 5.1: Optimization Stage
+
+Train the model on the optimization set. This is the only stage where the model sees data.
+
+- Run training with the chosen loss function.
+- Record optimization-set performance as a reference (this is expected to be the best performance — if evaluation-set performance exceeds it, something is wrong).
```
[the remark this answers is not in the document]
Comment (jmdkastro) [reply allora_research_model_skills/robustness-first/SKILL.md:281]: Not acting on this. Aware that eval > opt can occur for benign reasons (sample variance, regime quirks, easier eval window) — but the strong wording is intentional in a robustness-first skill. False positives (investigate when nothing was wrong) cost a builder ten minutes; false negatives (dismiss leakage as variance) wreck production. The skill is teaching a defensive habit: treat eval > opt as a trigger to investigate, not as a validation that the model is good. Softening the language would weaken the pedagogical signal that this skill specifically exists to deliver.

**Review by spooktheducks — APPROVED** [review]
